# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/PWebGL.ipdl
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/PWebGL.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/canvas/PWebGL.ipdl	Wed Jan 08 22:19:14 2020 +0000
@@ -0,0 +1,38 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * vim: sw=2 ts=8 et :
+ */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include protocol PCompositorBridge;
+include protocol PLayerTransaction;
+
+using mozilla::layers::CompositableHandle from "mozilla/layers/LayersTypes.h";
+
+namespace mozilla {
+namespace dom {
+
+/**
+ * Represents the connection between a WebGLChild actor that issues WebGL
+ * command from the content process, and a WebGLParent in the compositor
+ * process that runs the commands.
+ */
+sync protocol PWebGL
+{
+  manager PCompositorBridge;
+
+parent:
+  async __delete__();
+
+  // DLP: TODO: Does this need to be sync?
+  sync UpdateCompositableHandle(PLayerTransaction aLayerTrans,
+                         CompositableHandle aHandle);
+
+child:
+  // Tell client that this queue needs to be shut down
+  async QueueFailed();
+};
+
+} // dom
+} // mozilla